docs/reference/environment-variables.rst

```diff
@@ -0,0 +1,101 @@
+*********************
+Environment Variables
+*********************
+
+Applications built on craft-application have several environment variables that
+can configure their behaviour. They and the behaviour they modify are listed
+below.
+
+Variables passed to managed builders
+------------------------------------
+
+Several environment variables from the host environment are passed to the
+managed build environment. While an application may adjust these by adjusting
+the ``environment`` dictionary attached to the ``ProviderService``,
+craft-application will by default forward the ``http_proxy``, ``https_proxy``
+and ``no_proxy`` environment variables from the host.
+
+Supported variables
+-------------------
+
+These variables are explicitly supported for user configuration.
+
+.. _env-var-craft-build-environment:
+
+``CRAFT_BUILD_ENVIRONMENT``
+===========================
+
+Allows an environment to tell a craft application to run directly on the host
+rather than in managed mode if the value is ``host``. This is roughly equivalent
+to using ``--destructive-mode``, but is designed for configurations where the
+application is already being run in an appropriate container or VM, such as
+`Snapcraft rocks <https://github.com/canonical/snapcraft-rocks/>`_ or
+when controlled by a CI system such as `Launchpad <https://launchpad.net>`_.
+
+``CRAFT_BUILD_FOR``
+===================
+
+Sets the default architecture to build for. Is overridden by ``--build-for`` in
+lifecycle commands.
+
+``CRAFT_PLATFORM``
+==================
+
+Sets the default platform to build. Is overridden by ``--platform`` in lifecycle
+commands.
+
+``CRAFT_SNAP_CHANNEL``
+======================
+
+Overrides the default channel a craft application's snap is installed from if
+the manager instance is not running as a snap. If unset, the application will
+be installed from the specified channel.
+
+**NOTE:** If the application is running from a snap, the same revision as the
```

Comment:
We should consider notes as trivia that don't offer much of substance to the idea at hand. So, I'd like to avoid them whenever possible, because they have a tendency to break the document's flow. I can't tell whether this one is useful information. If it's trivial, could we remove it, and if not, can we make it a regular paragraph?